src/operator/tensor/dot-inl.cuh

```diff
@@ -35,6 +35,13 @@
 namespace mxnet {
 namespace op {
 
+// Returns integer log2(a) rounded up
+inline int log2i(size_t a) {
```

Comment:
I feel we should use common utility function with more efficient implementation for such type of computation.

Reply:
This is only used in this file, plus, pay attention to the mismatch between types of input and output.

Reply:
Although this is only used in this file for now, this function is too generic and very likely to be used in the future by other developers. Making it a generic utility will allow us to change its implementation in the future without breaking existing code. It's output can be cast based on our need. I don't see it as a reason to have a separate function to do that.

Reply:
IMO a generic utility should be a templated one to support all data types instead of something like this.

Reply:
I did a simple search and found log2i (or with slightly different names) function appeared at least three times in this module. See: indexing_op.h, pooled_storage_manner.h and intrinsics.cuh. It would be nice to consolidate them.

Reply:
For normal usages where the inputs and outputs have the same type there're std implementations of this. On the other hand this is already a very optimized code to perform such functionality.

Reply:
Taking log2 integer does sound like a common utility. We always need to calculate this before calling `SortByKey `. Can we have `log2i(size_t)` and `log2i(unsigned int)` in `common/utils.h` and replace other occurrences?